vue-component-decorator.ts

```diff
@@ -0,0 +1,101 @@
+// You should have `Vue` definitions downloaded using tsd
+// and here just reference your main tsd.d.ts file or
+// vue.d.ts to make the compilator work correctly.
+
+/// <reference path="../typings/tsd.d.ts"/>
+
+module Decorator {
+
+  var hooks = [
+      "created",
+      "ready",
+      "beforeCompile",
+      "compiled",
+      "beforeDestroy",
+      "destroyed",
+      "attached",
+      "detached"
+  ];
+
+  interface IOptions {
+    data?: {};
+    methods?: {};
+    computed?: {};
+  }
+
+  export function VueComponent(target: any) {
+
+      var options : IOptions = {};
+
+      // instance properties are data
+      var data = new target();
+
+      if (Object.keys(data).length) {
+        options.data = data;
```

Comment:
Here `options.data` needs to be a function that returns a fresh copy of the data. Otherwise the same object is shared across all instances of the class.